defaults/portal2/main.js

```diff
@@ -25,35 +25,102 @@ function wsMessageHandler (event) {
   const data = JSON.parse(event.data);
 
   switch (data.type) {
+    case "authenticated": {
+      ws.send(JSON.stringify({ type: "isGame", value: true }));
+      SendToConsole("clear; echo WebSocket connection established.");
+      return;
+    }
     case "cmd": return SendToConsole(data.value);
     case "ping": return ws.send(JSON.stringify({ type: "pong" }));
+    case "getMap": {
+      if (pathExists(`maps/${data.value.file}.bsp`)) {
+        ws.send(JSON.stringify({ type: "getMap", value: 1 }));
+      } else {
+        downloadMap(data.value.link, data.value.file);
+      }
+      return;
+    }
+    case "lobby_start" : {
+      menuSession = true;
+      SendToConsole("map " + data.map);
+      return;
+    }
   }
 
 }
 
 // Authenticates and sets up a WebSocket connection
-function wsSetup () {
+function wsSetup (token) {
+
+  const protocol = SERVER_ADDRESS.startsWith("https:") ? "wss" : "ws";
+  const hostname = SERVER_ADDRESS.split("://")[1].split("/")[0];
```

Comment:
This works.. but we do have the hostname in the configuration, so why not write that into address.txt?